src/codegen/templates/_library_interpreter.py.mako

```diff
@@ -543,17 +545,20 @@ class LibraryInterpreter(BaseInterpreter):
             bytes_per_chan_array,
         )
 
+        channel_array_length = channel_count * samples_read * number_of_signals_per_sample
+        channel_array = read_array[:channel_array_length].reshape(
+            channel_count, samples_read, number_of_signals_per_sample)
```

Comment:
Why not use GROUP_BY_SCAN_NUMBER, like the .NET API does? With that data layout, short reads are pre-squashed and DAQmx doesn't need to copy data around in order to squash them.

See https://github.com/ni/nidaqmx-python/issues/528 for details and examples.